src/Combinatorics/EnumerativeCombinatorics/types.jl

```diff
@@ -134,6 +134,51 @@ struct Partition{T<:IntegerUnion} <: AbstractVector{T}
   p::Vector{T}
 end
 
+# Iterator type
+struct Partitions{T<:IntegerUnion}
+  n::T
+
+  function Partitions(n::T) where T<:IntegerUnion
+    @req n >= 0 "n >= 0 required"
+    return new{T}(n)
+  end
+
+end
+
+# Iterator type
+# TODO: This should also have:
+#  - a flag for `only_distinct_parts`
+#  - support for upper/lower bounds
+struct PartitionsFixedNumParts{T<:IntegerUnion}
+  n::T
+  m::Int
+
+  lb::T
+  ub::T
+  distinct_parts::Bool
+
+  function PartitionsFixedNumParts(n::T, m::IntegerUnion, l1::IntegerUnion, l2::IntegerUnion, only_distinct_parts::Bool) where T<:IntegerUnion
+    @req n >= 0 "n >= 0 required"
+    @req m >= 0 "m >= 0 required"
+    @req l1 >= 0 "l1 >=0 required"
+    if l1 == 0
+      l1 = 1
+    end
+    return new{T}(n, convert(T, m), T(l1), T(l2), only_distinct_parts)
```

Comment:
The existing code used l1 and l2, though I think I prefer to use lb and ub - I can refactor to do this (while I'm at it, I will make it partitions of n into k parts instead of m parts).

Again following the preexisting code, we do not require that lb <= ub, but if this is not the case we return an empty iterator.